python/lsst/db/db.py

```diff
@@ -54,547 +41,115 @@
 ####################################################################################
 
 DbException = produceExceptionClass('DbException', [
-        (1500, "CANT_CONNECT_TO_DB", "Can't connect to database."),
-        (1505, "CANT_EXEC_SCRIPT",   "Can't execute script."),
-        (1510, "DB_EXISTS",          "Database already exists."),
-        (1515, "DB_DOES_NOT_EXIST",  "Database does not exist."),
-        (1520, "INVALID_CONN_INFO",  "Invalid connection parameter.") ,
-        (1525, "INVALID_DB_NAME",    "Invalid database name."),
-        (1530, "INVALID_OPT_FILE",   "Can't open the option file."),
-        (1532, "PASSWD_NOT_ALLOWED", "Password disallowed, use option file."),
-        (1535, "SERVER_CONNECT",     "Unable to connect to server."),
-        (1540, "SERVER_DISCONN",     "Failed to disconnect from db server."),
-        (1545, "SERVER_ERROR",       "Internal db server error."),
-        (1550, "NO_DB_SELECTED",     "No database selected."),
-        (1555, "NOT_CONNECTED",      "Not connected to the db server."),
-        (1560, "TB_DOES_NOT_EXIST",  "Table does not exist."),
-        (1565, "TB_EXISTS",          "Table already exists."),
-        (1900, "SERVER_WARNING",     "Warning."),
+        (1500, "CANT_EXEC_SCRIPT",   "Can't execute script."),
+        (1505, "DB_EXISTS",          "Database already exists."),
+        (1510, "DB_DOES_NOT_EXIST",  "Database does not exist."),
+        (1515, "INVALID_DB_NAME",    "Invalid database name."),
+        (1520, "MISSING_OPT_FILE",   "Can't open the option file."),
+        (1525, "INVALID_OPT_FILE",   "Invalid contents of option file."),
+        (1530, "PASSWD_NOT_ALLOWED", "Password disallowed, use option file."),
+        (1535, "TB_EXISTS",          "Table already exists."),
+        (1540, "TB_DOES_NOT_EXIST",  "Table does not exist."),
+        (1545, "UNSUPPORTED_DRIVER", "Unsupported driver, supported: mysql"),
+        (1550, "DRIVER_ERROR",       "Driver-specific error."),
+        (1555, "UNSAFE_STR",         "Unsafe string."),
         (9999, "INTERNAL",           "Internal error.")])
 
 ####################################################################################
+
 class Db(object):
     """
-    @brief Wrapper around MySQLdb.
+    @brief Thin wrapper around SQLAlchemy. It starts SQLAlchemy engine.
```

Comment:
This is not really a wrapper, it is now a factory class which created sqlalchemy engines. We should probably rename it altogether to something like `DbFactory`. I also think that we need better naming for module itself, `from lsst.db.db import Db` looks a bit ugly.